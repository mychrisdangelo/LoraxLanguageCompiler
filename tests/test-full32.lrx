/* Tests float operations
    Modified from Stephen Edwards' MicroC
*/
int main()
{
  print(1.0 + 2.0);
  print(1.0 - 2.0);
  print(1.0 * 2.0);
  print(135.0 / 2.0);
  print(99.0);
  print(1.0 == 2.0);
  print(1.0 == 2.0);
  print(1.0 == 1.0);
  print(99.0);
  print(1.0 != 2.0);
  print(1.0 != 1.0);
  print(99.0);
  print(1.0 < 2.0);
  print(2.0 < 1.0);
  print(99.0);
  print(1.0 <= 2.0);
  print(1.0 <= 1.0);
  print(2.0 <= 1.0);
  print(99.0);
  print(1.0 > 2.0);
  print(2.0 > 1.0);
  print(99.0);
  print(1.0 >= 2.0);
  print(1.0 >= 1.0);
  print(2.0 >= 1.0);
}
