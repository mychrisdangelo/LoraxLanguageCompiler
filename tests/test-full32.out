3.000000-1.0000002.00000067.50000099.000000falsefalsetrue99.000000truefalse99.000000truefalse99.000000truetruefalse99.000000falsetrue99.000000falsetruetrue